### ISSUE TITLE
Email newsletter form validation errors are not announced

### ISSUE THREAD

**Issue by alisonrae** [PublicisSapient/accessible-ecommerce-demo#41]
Default HTML5 error validation only works for a sighted user. Please try to make it work this way:

1. User submits form with error message
2. Error message appears visually near form field
3. Error message is inserted within the field's label. E.g.

Original label: 
`<label for="mylabel">Enter your email address</label><input id="mylabel"... />`

New label after submitting form with invalid email address:
`<label for="mylabel">Enter your email address <span class="error">Please enter a valid email address</span></label>`

The part in the `<span>` gets added programmatically.
4. Programmatically put the user's keyboard focus into the field that has the first error (in this case there is only one field so it is easy). The screen reader will read the field AND the label.

**Comment by mikequattrin** [comment]
Tested and marking as resolved. Excellent work, Alan!

**Comment by alisonrae** [comment]
Reopening this one, because the error message that is shown to a sighted person is not read to a non-sighted person. The very same message has to be read out loud. It's not enough for the browser to say "Invalid data edit text". Please make it work exactly as described in the original ticket. As noted, I never saw html5 error messaging work in an accessible way. You may have to ditch that and do custom error messaging. See accessible error validation at work here: https://codepen.io/lsnrae/pen/pdBRNo

**Comment by mikequattrin** [comment]
@aevanson @alisonrae Apologies for closing prematurely. I tested this locally, as the live site automagically updates and we have no control over when this occurs. Alan has added custom error handling that resolves this issue once the page is updated.
